buildflow/api/storage.py

```diff
@@ -0,0 +1,30 @@
+from typing import Callable, Dict, Optional, Protocol
+
+from buildflow.api.providers import Provider
+
+
+# NOTE: This class is a protocol for dataclasses to catch static typing errors.
+class DataclassProtocol(Protocol):
+    __dataclass_fields__: Dict
+    __dataclass_params__: Dict
+    __post_init__: Optional[Callable]
+
+
+# NOTE: This class represents a single table.
```

Comment:
Should this abstraction be table or database level?

Mostly people will likely have one database that has many tables that they may need to read or write from. Ideally you maintain one DB connection that is able to read from all tables.

Reply:
Ah nvm I see how it works now with the examples below. One provider is used across multiple decoractors. Maybe add a comment mentioning that